src/main/java/core/basesyntax/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.service.operation;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public int getQuantity(int defaultQuantity, int fruitQuantity) {
+        return defaultQuantity - fruitQuantity;
+    }
+}
```

Comment:
The implementation of 'PurchaseOperation' does not check if the result balance is positive before saving it to the Storage. According to the checklist, you should validate that the resulting balance after a purchase operation is not negative, and throw a RuntimeException if it is.